File: AdminP/AdminP.Web/Modules/ServerTypes/Default/ReviewForm.ts
import { initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface ReviewForm {
    Name: StringEditor;
    Email: StringEditor;
    Subject: StringEditor;
    Description: StringEditor;
}

export class ReviewForm extends PrefixedContext {
    static readonly formKey = 'Default.Review';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ReviewForm.init) {
            ReviewForm.init = true;

            var w0 = StringEditor;

            initFormType(ReviewForm, [
                'Name', w0,
                'Email', w0,
                'Subject', w0,
                'Description', w0
            ]);
        }
    }
}

File: AdminP/AdminP.Web/Modules/Default/Review/ReviewForm.cs
namespace AdminP.Default.Forms;

[FormScript("Default.Review")]
[BasedOnRow(typeof(ReviewRow), CheckNames = true)]
public class ReviewForm
{
    public string Name { get; set; }
    public string Email { get; set; }
    public string Subject { get; set; }
    public string Description { get; set; }
}